src/world/mod.rs

```diff
@@ -16,11 +16,67 @@ mod lazy;
 #[cfg(test)]
 mod tests;
 
+#[cfg(not(feature = "nightly"))]
+/// Note: Enable nightly feature to get the component type
+/// printed instead of just a vague catch all message.
```

Comment:
I'm not sure how this doc comment helps a user getting a panic? I thought we could include this note in the string.